fop/fop.h

```diff
@@ -306,7 +306,12 @@ void m0_fop_type_init(struct m0_fop_type *ft,
  */
 #define M0_FOP_TYPE_INIT(ft, ...)                                        \
 	m0_fop_type_init((ft), &(const struct __m0_fop_type_init_args) { \
-			__VA_ARGS__ })
+			__VA_ARGS__ });
+
+#define M0_FOP_TYPE_INIT2_DECL struct __m0_fop_type_init_args __args
+#define M0_FOP_TYPE_INIT2(ft, ...) ({                                    \
+	__args = (struct __m0_fop_type_init_args) {__VA_ARGS__};         \
+	m0_fop_type_init((ft), &__args); })
```

Comment:
Can you explain why this is needed?

Why is this needed only in m0_mdservice_fopts_init() and m0_mdservice_rep_fopts_init()?
Not needed in other places where M0_FOP_TYPE_INIT() is used.

Reply:
It reuses the same `struct __m0_fop_type_init_args __args` object in each `M0_FOP_TYPE_INIT2()` call instead of creating it again and again (on the stack!) in each `M0_FOP_TYPE_INIT()` call.

I should probably add some comment explaining this.

Reply:
Added a comment, please check if it helps.

Reply:
>It reuses the same struct __m0_fop_type_init_args __args object in each M0_FOP_TYPE_INIT2() call 
I understand this.

>instead of creating it again and again (on the stack!) in each M0_FOP_TYPE_INIT() call.

If we use the macro M0_FOP_TYPE_INIT(), it will create an anonymous "struct __m0_fop_type_init_args" variable and pass it to the function m0_fop_type_init()?

What about the invoking of this M0_FOP_TYPE_INIT() in other places?
They are not used in kernel space, so it does not matter?

Reply:
Yes. About other places - they might be used in kernel space also, but within the 1K limit, so it's OK.